### PR TITLE
Restructured Scope Docs

### PR DESCRIPTION
**Pull request by abuehrle**
Here's a first draft of the docs for the .15 release.  Still to do:

- [ ] ECS AMIs
- [x] incorporate changes from https://github.com/weaveworks/scope/pull/1426
- [x] incorporate changes from https://github.com/weaveworks/scope/pull/1434

Punted for now:
- Redraw the architecture drawings, ACII doesn't look great in WP
- Add sample app that describes how to troubleshoot issues (if there is time)
- Screen captures and description of Views, filtering, and metrics <--- the new grid view here as well?
- Update kubernetes namespace filters (are they implemented as filters or as views?)
- Weave Net integration?

Please have a look and let me if the structure is ok, and any other topics you think I should add.

### CODE REVIEW

File: site/integrations.md [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+Launch Docker Compose with one of the two fragments below and set the value of the Scope as a Service token in an environment variable:
+
+
+    SCOPE_SERVICE_TOKEN=abcdef_my_token  docker-compose up -d
+
+
+###Docker Compose Format Version 1:
+
+    probe:
+      image: weaveworks/scope:0.15
```
Comment (2opremio) [inline site/integrations.md]: It's going to be a bit of a pain to update this with every release. I think this should live in github and be referenced through a link.

File: site/integrations.md [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+        - "true"
+        - "--service-token"
+        - "${SCOPE_SERVICE_TOKEN}"
+
+##Docker Compose Format Version 2:
+
+    version: '2'
+    services:
+      probe:
+        image: weaveworks/scope:0.15
```
Comment (2opremio) [inline site/integrations.md]: Same here

File: site/integrations/aws-cloud-formation.md [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+Before launching the stack:
+
+* [Set up an Amazon Account](http://docs.aws.amazon.com/AWSEC2/latest/UserGuide/get-set-up-for-amazon-ec2.html)
+* [Create the Key Pairs]("http://docs.aws.amazon.com/gettingstarted/latest/wah/getting-started-prereq.html). You will need to reference the name of the key pairs when you create the stack.
+
+The link below will launch a sample app using a Cloudformation template, but you can swap out the IdentiOrca app and use your own app instead. 
+
+**Ready to launch a stack?  Click here to launch a stack to AWS:**
+
+![](cloudformation-launch-stack.png)[1]
```
Comment (2opremio) [inline site/integrations/aws-cloud-formation.md]: 1?

**Comment by 2opremio** [comment]
I think we should remove the browser navigation bar from the screenshots.

File: site/integrations/aws-cloud-formation.md (hunk 3 above)
Comment (abuehrle) [inline site/integrations/aws-cloud-formation.md]: I couldn't figure out how get a link on an image with markdown.  Do you know how to do this?

File: site/integrations.md (hunk 2 above)
Comment (abuehrle) [inline site/integrations.md]: Good point.  Where should these live? In examples?

File: site/integrations/aws-cloud-formation.md (hunk 3 above)
Comment (2opremio) [inline site/integrations/aws-cloud-formation.md]: Doesn't  `[](https://www.weave.works/deploy-weave-aws-cloudformation-template/)` work?

File: site/integrations.md (hunk 2 above)
Comment (2opremio) [inline site/integrations.md]: I have just checked and they already seem to be in the examples, we just need to link there.

Comment (abuehrle) [inline site/integrations.md]: I'm going to keep this in here.  Too much clicking around for the user. I think it's best to keep the docs with the docs, and not send the user to 2 different places for answers.

Comment (2opremio) [inline site/integrations.md]: Fair enough, but it's likely to get outdated.

Comment (abuehrle) [inline site/integrations.md]: Yes, that's why the docs need to be updated with every release. You guys are already doing that with the READme.md so what's the difference?

**Comment by tomwilkie** [comment]
I'm keen to get this merged ASAP to stop the existing readme getting out of sync with the docs.  With this in mind, can we:
- [ ] punt unfinished sections to a future PR? (I'm thinking app-troubleshooting.md, managing-views.md)
- [ ] use the existing screenshots in `imgs` for now?
- [ ] use existing ascii art diagrams?
- [ ] focus on existing 0.14 features?

**Comment by tomwilkie** [comment]
The distinction between an integration & how to install scope is not clear to me from these docs.  Most of what is classed as an integration really appears to be instructions on how to install scope on a given platform.  Can we have a single file (installation.md) with a section each for Docker (subsection for compose), Kubernetes, ECS, DC/OS and Rancher?

File: site/building.md [hunk 4]
```diff
@@ -0,0 +1,82 @@
+---
+title: Building and Developing
```
Comment (tomwilkie) [inline site/building.md]: Title is building and developing but sections are Developing and Debugging.  I think these should be consistent.

File: site/building.md [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+[Go](https://golang.org).
+
+## <a name="debugging"></a>Debugging
+
+Scope has a collection of built-in debugging tools to aid Scope developers.
+
+
+To get debug information in the logs launch Scope with `--debug`:
+
+    scope launch --debug
```
Comment (tomwilkie) [inline site/building.md]: We seem to have lost the code fence formatting here.

File: site/building.md [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+    docker logs weavescope
+
+
+To have the Scope App or Scope Probe dump their goroutine stacks, run:
+
+    kill -QUIT $(pgrep -f scope-(app|probe))
```
Comment (tomwilkie) [inline site/building.md]: Formatting

File: site/building.md [hunk 7]
```diff
@@ -0,0 +1,82 @@
...
+    docker logs weavescope
+
+
+The Scope Probe is instrumented with various counters and timers. To have it dump those values, run:
+
+
+    kill -USR1 $(pgrep -f scope-probe)
```
Comment (tomwilkie) [inline site/building.md]: Formatting

File: site/building.md [hunk 8]
```diff
@@ -0,0 +1,82 @@
...
+    docker logs weavescope
+
+ Both the Scope App and the Scope Probe offer [HTTP endpoints with profiling information](https://golang.org/pkg/net/http/pprof/).
+
```
Comment (tomwilkie) [inline site/building.md]: Spurious new line?

File: site/building.md [hunk 9]
```diff
@@ -0,0 +1,82 @@
...
+These cover things such as CPU usage and memory consumption:
+
+  * The Scope App enables its HTTP profiling endpoints by default, which are accessible on the same port the Scope UI is served (4040).
+  * The Scope Probe doesn't enable its profiling endpoints by default. To enable them, you must launch Scope with `--probe.http.listen addr:port`. For instance, launching Scope with `scope launch --probe.http.listen :4041`, will allow you access the Scope Probe's profiling endpoints on port 4041.
+
+Then, you can collect profiles in the usual way. For instance:
+
+To collect the memory profile of the Scope App:
+
+    go tool pprof http://localhost:4040/debug/pprof/heap
```
Comment (tomwilkie) [inline site/building.md]: Formatting

File: site/building.md [hunk 10]
```diff
@@ -0,0 +1,82 @@
...
+
+To collect the CPU profile of the Scope Probe:
+
+    go tool pprof http://localhost:4041/debug/pprof/profile
```
Comment (tomwilkie) [inline site/building.md]: Formatting

File: site/building.md [hunk 11]
```diff
@@ -0,0 +1,82 @@
...
+
+If you don't have `go` installed, you can use a Docker container instead:
+
+To collect the memory profile of the Scope App:
+
+    docker run --net=host -v $PWD:/root/pprof golang go tool pprof http://localhost:4040/debug/pprof/heap
```
Comment (tomwilkie) [inline site/building.md]: Formatting

File: site/building.md [hunk 12]
```diff
@@ -0,0 +1,82 @@
...
+
+To collect the CPU profile of the Scope Probe:
+
+    docker run --net=host -v $PWD:/root/pprof golang go tool pprof http://localhost:4041/debug/pprof/profile
```
Comment (tomwilkie) [inline site/building.md]: Formatting

File: site/how-it-works.md [hunk 13]
```diff
@@ -0,0 +1,44 @@
...
+menu_order: 70
+---
+
+The following topics are discussed: 
+
+* [Scope Reporting in Service Mode](#scope-reporting-in-service-mode)
+* [Scope Reporting in Stand-alone Mode](#scope-reporting-in-stand-alone-mode)
+* [Managing Clusters with Scope and weaveDNS](#managing-clusters-with-scope-and-weavedns)
+
+Weave Scope consists of two components: the app and the probe. The components are deployed as a single Docker container using the scope script. The probe is responsible for gathering information about the host on which it is running. This information is sent to the app in the form of a report. The app processes reports from the probe into usable topologies, serving the UI, as well as pushing these topologies to the UI.
```
Comment (tomwilkie) [inline site/how-it-works.md]: By default, the components are deployed...

File: site/installing-scope.md [hunk 14]
```diff
@@ -0,0 +1,75 @@
...
+
+Scope needs to be installed onto every machine that you want to monitor. Once launched, Scope doesn’t require any other configuration and it also doesn’t depend on Weave Net.
+
+The script downloads and runs the latest released Scope image from the Docker Hub. After it’s been launched, open your browser to `http://localhost:4040`.
+
+>**Note:** If you're using boot2docker, replace localhost with the output of boot2docker IP.
+
+>  Scope allows anyone with access to the UI control over your containers: as such, the Scope app endpoint (port 4040) should not be made accessible on the Internet.  Additionally traffic between the app and the probe is currently insecure and should also not traverse the Internet.
+
+###Disabling Automatic Updates
```
Comment (tomwilkie) [inline site/installing-scope.md]: I would move this to how-it-works.md

File: site/building.md (hunk 5 above)
Comment (abuehrle) [inline site/building.md]: I've put four spaces in for code formatting.  The back ticks in WP are rendered with a shadowed single line, whereas the spaces will put a nice shaded box around the code.

Comment (tomwilkie) [inline site/building.md]: Ah my bad, sorry.

File: site/introducing-weave-scope.md [hunk 15]
```diff
@@ -0,0 +1,68 @@
+---
+title: Introducing Weave Scope
+menu_order: 10
+---
+
+
+Weave Scope is a browser-based infrastructure, app diagnostics and container visualization tool. It provides a top down view into your app as well as your entire infrastructure, and allows you to diagnose any problems with your distributed containerized app, in real time, as it being deployed to a cloud provider.
```
Comment (tomwilkie) [inline site/introducing-weave-scope.md]: The tag line we've been using is "Weave Scope: Monitoring, visualisation & management for Docker & Kubernetes".  How does this introduction fit with that?

Comment (tomwilkie) [inline site/introducing-weave-scope.md]: Also I think its could be misleading to use the term "browser-based" - the majority of Scope is server side.

File: site/introducing-weave-scope.md [hunk 16]
```diff
@@ -0,0 +1,68 @@
...
+---
+
+
+Weave Scope is a browser-based infrastructure, app diagnostics and container visualization tool. It provides a top down view into your app as well as your entire infrastructure, and allows you to diagnose any problems with your distributed containerized app, in real time, as it being deployed to a cloud provider.  
+
+Scope can be used in standalone mode, where you deploy it locally onto your hosts, or you can use Weave Scope as a Service, see [Installing Weave Scope](/site/installing-scope.md) for more information. 
+
+##Getting Microservices Under Control
+
+Microservices-based architecture poses significant challenges when deployed to Docker containers on the cloud. But with Weave Scope, visualizing network bottlenecks, troubleshooting CPU consumption and memory leaks is simplified. With Views, you can quickly examine various metrics about your containerized app.
```
Comment (tomwilkie) [inline site/introducing-weave-scope.md]: > Microservices-based architecture poses significant challenges when deployed to Docker containers on the cloud

They do?  Like what?

File: site/introducing-weave-scope.md [hunk 17]
```diff
@@ -0,0 +1,68 @@
...
+
+See, [Troubleshooting Containerized Apps](/site/app-troubleshooting.md).
+
+##Orchestrator Aware and Cloud Integrations
+
+Weave Scope can visualize your app within: 
+
+ * Docker Compose
+ * Kubernetes
+ * AWS Cloud Formation
```
Comment (tomwilkie) [inline site/introducing-weave-scope.md]: Docker Compose & AWS Cloud Formation seems misplaced here.  Cloud formation & componse is a tool for deployment, not a runtime as implied by "within".

I would go with Docker (optionally Docker Datacenter or Docker Swarm), Kubernetes, and Amazon ECS

File: site/introducing-weave-scope.md [hunk 18]
```diff
@@ -0,0 +1,68 @@
...
+
+
+For instructions on Weave Scope integrations, see [Integrations](/site/integrations.md).
```
Comment (tomwilkie) [inline site/introducing-weave-scope.md]: No more integrations section.

File: site/building.md [hunk 19]
```diff
@@ -0,0 +1,82 @@
...
+
+ * [Developing](#developing)
+ * [Debugging](#debugging)
+
+
+## <a name="developing"></a>Developing
+
+Building Scope from source depends on the latest version of [docker](https://www.docker.com/), so please install that before proceeding.
+
+The main build is `make`, which builds the UI build container, builds the UI in said container, builds the backend build container, builds the app and probe in said container, and finally pushes the lot into a Docker image called **weaveworks/scope**.
```
Comment (tomwilkie) [inline site/building.md]: > The main build is `make`

This reads weirdly.  The main build tool is make?  The build is orchestrated by make?  The build process is automated using make?

File: site/building.md [hunk 20]
```diff
@@ -0,0 +1,82 @@
...
+    docker logs weavescope
+
+
+The Scope Probe is instrumented with various counters and timers. To have it dump those values, run:
+
+
+    kill -USR1 $(pgrep -f scope-probe)
+    docker logs weavescope
+
+ Both the Scope App and the Scope Probe offer [HTTP endpoints with profiling information](https://golang.org/pkg/net/http/pprof/).
```
Comment (tomwilkie) [inline site/building.md]: I'd be tempted to stick a new heading here "Profilling"

File: site/installing-scope.md [hunk 21]
```diff
@@ -0,0 +1,175 @@
...
+The following topics are discussed: 
+
+ * [Weave Scope Cloud Service](#signing-up-for-weave-scope-cloud-service)
+ * [Launching Scope as a Service with Docker Compose](#launching-scope-as-a-service-and-docker-compose)
+ * [Launching a Stack with Scope to AWS](#launching-a-stack-with-scope-to-aws)
+ * [Installing Scope as a DC/OS Package](#installing-scope-as-a-dc/os-package)
+ * [Visualizing Kubernetes Clusters with Scope](#visualizing-kubernetes-clusters-with-scope)
+
+
+##<a name="signing-up-for-weave-scope-cloud-service"></a>Signing Up for Weave Scope as a Service
```
Comment (tomwilkie) [inline site/installing-scope.md]: I though we'd agreed not to use the term "Scope as a Service"?

File: site/installing-scope.md [hunk 22]
```diff
@@ -0,0 +1,175 @@
...
+title: Installing Weave Scope
+menu_order: 20
+---
+
+
+Weave Scope consists of three parts: the probe, the app and the user interface. It can be deployed either as a standalone configuration, or if you don’t want to manage the administration yourself, you can sign up for Weave Scope as a service.
+
+The following topics are discussed: 
+
+ * [Weave Scope Cloud Service](#signing-up-for-weave-scope-cloud-service)
```
Comment (tomwilkie) [inline site/installing-scope.md]: Title of heading does not match list item.

Comment (tomwilkie) [inline site/installing-scope.md]: In general I think the structure should be:

```
* Installing Scope on Docker
  * Using Docker Compose
  * Using the Scope Service
* Installing Scope on Kubernetes
  * Using the Scope Service
* Installing Scope on Amazon ECS
  * Using the Scope Service
* Installing Scope on Mesosphere DC/OS
  * Using the Scope Service
```

I appreciate we want to funnel people towards the service at some point, but its not ready from prime time yet.  Lets make the top code block on the page the instructions on how to launch scope locally against docker, which is what we suspect the vast majority of users will need.

File: site/installing-scope.md [hunk 23]
```diff
@@ -0,0 +1,175 @@
...
+    sudo scope launch --service-token=<token>
+
+Where, 
+
+`--service-token=<token>`  is the token that was sent to you when you signed 
+up.
+
+Once the probes have been launched, log into the Scope service, and click on 'My Scope' in the top right-hand corner. 
+
+###Disabling Automatic Updates
```
Comment (tomwilkie) [inline site/installing-scope.md]: This section seems misplaced here (it is not service specific).  I'd stick it at the bottom of this doc for now.

File: site/scope-plugins.md [hunk 24]
```diff
@@ -0,0 +1,80 @@
+---
+title: Generating Custom Metrics with Plugins
+menu_order: 80
+---
+
+The following topics are discussed:
```
Comment (tomwilkie) [inline site/scope-plugins.md]: This could go in the developing section, as its primarily targeted at developers.

File: site/how-it-works.md [hunk 25]
```diff
@@ -0,0 +1,60 @@
+---
+title: Understanding Weave Scope
+menu_order: 70
+---
+
+The following topics are discussed: 
+
+* [Scope Reporting in Service Mode](#scope-reporting-in-service-mode)
+* [Scope Reporting in Standalone Mode](#scope-reporting-in-stand-alone-mode)
+* [Installing Weave Scope in Standalone Mode](#installing-weave-scope-standalone)
```
Comment (tomwilkie) [inline site/how-it-works.md]: Installation instruction seem out of place here.

File: site/how-it-works.md [hunk 26]
```diff
@@ -0,0 +1,60 @@
...
+* [Managing Clusters with Scope and weaveDNS](#managing-clusters-with-scope-and-weavedns)
+
+Weave Scope consists of two components: the app and the probe. The components are deployed as a single Docker container using the scope script. The probe is responsible for gathering information about the host on which it is running. This information is sent to the app in the form of a report. The app processes reports from the probe into usable topologies, serving the UI, as well as pushing these topologies to the UI.
+
+
+## <a name="scope-reporting-in-service-mode"></a>Scope Reporting in Service Mode
```
Comment (tomwilkie) [inline site/how-it-works.md]: The term "reporting" seems weird here.

File: site/how-it-works.md [hunk 27]
```diff
@@ -0,0 +1,60 @@
...
+
+Scope needs to be installed onto every machine that you want to monitor. Once launched, Scope doesn’t require any other configuration and it also doesn’t depend on Weave Net.
+
+The script downloads and runs the latest released Scope image from the Docker Hub. After it’s been launched, open your browser to `http://localhost:4040`.
+
+>**Note:** If you're using boot2docker, replace localhost with the output of boot2docker IP.
+
+>  Scope allows anyone with access to the UI control over your containers: as such, the Scope app endpoint (port 4040) should not be made accessible on the Internet.  Additionally traffic between the app and the probe is currently insecure and should also not traverse the Internet.
+
+##<a name="managing-clusters-with-scope-and-weavedns"></a>Managing Clusters with Scope and weaveDNS
```
Comment (tomwilkie) [inline site/how-it-works.md]: This also belongs in the installation section, under "Installing Scope on Docker" > "Using Weave Net"

File: site/how-it-works.md [hunk 28]
```diff
@@ -0,0 +1,60 @@
...
+
+!['Service Mode Reporting'](service-mode.png)
+
+##<a name="scope-reporting-in-stand-alone-mode"></a>Scope Reporting in Standalone Mode
+
+When running Scope in a cluster, each probe sends its reports to a dedicated app. The app merges the reports from its probe into a comprehensive report that is sent to the browser.  To visualize your entire infrastructure and apps running on that infrastructure, Scope must be launched on to every machine you are using.
+
+!['Stand-alone Reporting'](stand-alone.png)
+
+##<a name="installing-weave-scope-standalone"></a>Installing Weave Scope Standalone
```
Comment (tomwilkie) [inline site/how-it-works.md]: I would move this to the introduction as this is what most people will be looking for.

I would add a line at the bottom of this section, in the introduction, saying "for instructions on how to install scope on other platforms, see..."

File: site/scope-features.md [hunk 29]
```diff
@@ -0,0 +1,68 @@
...
+Interact with your containers directly: pause, restart and stop containers without having to leave the Scope browser window.
+
+##<a name="troubleshoot-apps"></a>Troubleshoot Apps
+
+A convenient terminal window is provided that enables you to interact with your app and to troubleshoot and diagnose any issues all within the same context. 
+
+
+##<a name="weave-net-integration"></a>Weave Net Integration
+
+Visualize Weave Net as it gets deployed across your cloud environment, troubleshoot services and manage your networked containers all from a convenient graphical user interface.
```
Comment (tomwilkie) [inline site/scope-features.md]: We don't actually do this.

**Comment by tomwilkie** [comment]
@abuehrle looking good.  I've been quite direct in my feedback (sorry!), try and read it as if I was being nice.

**Comment by tomwilkie** [comment]
Assign this back to me when you're happy with it.

File: site/how-it-works.md (hunk 27 above)
Comment (abuehrle) [inline site/how-it-works.md]: yes, I thought that as well.

File: site/how-it-works.md (hunk 25 above)
Comment (abuehrle) [inline site/how-it-works.md]: I thought that as well. But you asked me to move it there.

**Comment by abuehrle** [comment]
@tomwilkie  I am not sure what you mean for the install page, so maybe it is better if you do that. I have addressed some items in the overview section. Still to add are screen captures, new feature (search), and whatever was added to the readme since this branch has been open.

**Comment by tomwilkie** [comment]
@abuehrle All done.  Before these are merged, they will need squashing (let me know if you need a hand).

**Comment by abuehrle** [comment]
I think this is ready to squash now, and then we'll fix up the readme and also make sure we haven't missed anything  @tomwilkie

**Pull request merged**